Removing CLANG_MODULE setting.

nmahendru opened the pull request:
#### 55c18ec706cda282dcf3b1c87e84db57cdfd5ac2
<pre>
Removing CLANG_MODULE setting.
<a href="https://rdar.apple.com/124559419">rdar://124559419</a>

Reviewed by Elliot Williams.

Looks like it&apos;s not really needed for Swift interop.
Newer versions of clang are causing the build to fail otherwise.

Error:
In file included from /Volumes/Data/safari-4/OpenSource/Source/WebCore/PAL/pal/text/TextCodecLatin1.cpp:31:
/Volumes/Data/safari-4/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/text/CString.h:77:17: error: definition of &apos;basic_string&apos; must be imported from module &apos;std_string&apos; before it is required
   77 |     std::string toStdString() const { return m_buffer ? std::string(m_buffer-&gt;data()) : std::string(); }
      |
* Source/WebCore/PAL/Configurations/PAL.xcconfig:

Canonical link: <a href="https://commits.webkit.org/276112@main">https://commits.webkit.org/276112@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cd016b0476406f7a16fa519695c22a3a6ae1a758

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/43651 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/22699 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/46089 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/46294 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/39777 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/45955 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/26523 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/20107 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/36065 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/44225 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/19713 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/37603 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/17042 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/17261 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/38665 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/1708 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/39807 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/38972 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/47844 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/18690 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/15284 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/42857 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/20108 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/41531 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9736 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/20289 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/19737 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple reviewed (approved):
Sounds good to me, but could you please associate a bug with this before landing?

nmahendru committed:
Removing CLANG_MODULE setting.
rdar://124559419

Reviewed by Elliot Williams.

Looks like it's not really needed for Swift interop.
Newer versions of clang are causing the build to fail otherwise.

Error:
In file included from /Volumes/Data/safari-4/OpenSource/Source/WebCore/PAL/pal/text/TextCodecLatin1.cpp:31:
/Volumes/Data/safari-4/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/text/CString.h:77:17: error: definition of 'basic_string' must be imported from module 'std_string' before it is required
   77 |     std::string toStdString() const { return m_buffer ? std::string(m_buffer->data()) : std::string(); }
      |
* Source/WebCore/PAL/Configurations/PAL.xcconfig:

Canonical link: https://commits.webkit.org/276112@main
